include/zephyr/net/dhcpv4.h

```diff
@@ -25,6 +25,60 @@ extern "C" {
  * @{
  */
 
+struct net_dhcpv4_option_callback;
+
+/**
+ * @typedef net_dhcpv4_option_callback_handler_t
+ * @brief Define the application callback handler function signature
+ *
+ * @param option The DHCP option tag
+ * @param cb Original struct net_dhcpv4_option_callback owning this handler
+ * @param length The length of data received from the server
+ *
+ * Note: cb pointer can be used to retrieve private data through
+ * CONTAINER_OF() if original struct net_dhcpv4_option_callback is stored in
+ * another private structure.
+ */
+typedef void (*net_dhcpv4_option_callback_handler_t)(const uint8_t option,
```

Comment:
Isn't `option` redundant here? It's already part of the `struct net_dhcpv4_option_callback`

Reply:
IMO it'd make sense to include `struct net_if *` pointer instead, in case there are multiple network interfaces in the system.

Reply:
Good points, I will address them